### ISSUE TITLE
Network ID as a top level module output?

### ISSUE THREAD

**Issue by alexjmoore** [terraform-google-modules/terraform-google-network#307]
It would be nice to get the network ID as a top level output from the module.  For example I had to do this to get the ID:

`module.network["vpc-test"].network.network.id]`

Would be nice to have:

`module.network["vpc-test"].network_id]`

**Comment by bharathkkb** [comment]
Hi @alexjmoore 
We could add this but most use cases I have encountered seem to use `network_self_link`. What is your use case for `network.id`?

**Comment by alexjmoore** [comment]
It's a good point, because I did think it unusual to have to use it.  It's in the `google_active_directory_domain` resource, eg:
```
resource "google_active_directory_domain" "ad-domain" {
  domain_name       = "ad1.example.com"
  locations         = ["europe-west2"]
  reserved_ip_range = "192.168.255.0/24"
  authorized_networks = [module.network["vpc-test"].network.network.id]
}
```
It actually confused me quite a bit getting it to work as the docs for that resource aren't super clear what you put there, initially I tried just the network name, then the self_link, in fact even on the gcloud cli the network name doesn't work....I suspect its an inconsistency on the API itself.

Thoughts?

**Comment by morgante** [comment]
Since usage of the network ID is quite rare and there *is* a solution here, I'm inclined to think we shouldn't add an additional output.

**Comment by alexjmoore** [comment]
Is it worth raising a case against the provider itself to see if it makes sense to revise resource google_active_directory_domain to take the self link you think?

**Comment by morgante** [comment]
Sure, I think that would be a reasonable request since most resources already accept self-links.

**Comment by alexjmoore** [comment]
@morgante it seems the provider folks indicate that id is the preferred field, as per: https://github.com/hashicorp/terraform-provider-google/issues/9546  thoughts?

**Comment by morgante** [comment]
Hmm, I think we can add this as an output then.

**Comment by 3sne** [comment]
Hi, I have submitted a PR for this issue.

**Comment by morgante** [comment]
Fixed by #315
